static/js/composebox_typeahead.js

```diff
@@ -315,9 +317,9 @@ exports.content_typeahead_selected = function (item) {
     if (this.completing === 'emoji') {
         // leading and trailing spaces are required for emoji, except if it begins a message.
         if (beginning.lastIndexOf(":") === 0 || beginning.charAt(beginning.lastIndexOf(":") - 1) === " ") {
-            beginning = beginning.replace(/:\S+$/, "") + ":" + item.emoji_name + ": ";
+            beginning = beginning.replace(/:\S+\s?[a-zA-Z0-9]*$/, "") + ":" + item.emoji_name + ":";
```

Comment:
You probably just want something like `beginning = (beginning.substring(0, beginning.length - this.token.length-1)`, similar to the other cases below? There are emoji with multiple spaces, and other characters, like `-`. 
Also, seems like you lost the space at the end, after the `: `.